frontend/src/components/CredentialTile.vue

```diff
@@ -115,19 +127,35 @@ export default {
       return undefined
     },
     phase () {
-      if (!this.type) {
-        return this.shootStatusCredentialsRotationAggregatedPhase
+      if (this.type === 'ALL_CREDENTIALS') {
+        return this.shootStatusCredentialsRotationAggregatedPhase ?? {}
       }
-      const type = this.rotationStatus.phase
+
       return {
-        type
+        type: this.rotationStatus.phase
       }
     },
+    phaseType () {
+      if (!this.phase) {
+        return undefined
+      }
+      if (this.phase.type) {
+        return this.phase.type
+      }
+      if (this.rotationStatus.lastInitiationTime > (this.rotationStatus.lastCompletionTime ?? '0')) {
```

Comment:
May be clearer:
```js
if (!this.rotationStatus.lastCompletionTime || this.rotationStatus.lastInitiationTime > this.rotationStatus.lastCompletionTime)
```
What if lastInitiationTime is undefined?

Reply:
we want to compare the timestamps here. If `lastInitiationTime` is greater than `lastCompletionTime` this one-step rotation is currently Rotating.
If `lastInitiationTime` is undefined the if statement is `false`. if lastCompletionTime is undefined we need to default to `'0'` because `lastInitiationTime > undefined` would not result to `true`

Reply:
Handle the undefined cases explicitly. Something like this:
```js
if (!this.rotationStatus.lastCompletionTime || (
      this.rotationStatus.lastInitiationTime && 
      this.rotationStatus.lastInitiationTime > this.rotationStatus.lastCompletionTime
    )) {
    ...
}
```